File: rules.js
// field constants
var FIELD = {}
FIELD.PK_WITH_PREFIX = 0

//table constants
var TABLE = {};
TABLE.SKIP = 0
TABLE.TRUNCATE = 1

var RULES = {
    "constants": {
        "table.skip": TABLE.SKIP,
        "field.pk_with_prefix": FIELD.PK_WITH_PREFIX,
    },
    "tables": {
        "public.main_contragent": {
            "first_name": {"type": FIELD.PK_WITH_PREFIX, "prefix": "i"},
            "passport_seria": {"type": FIELD.PK_WITH_PREFIX, "zerofill": 4},
            "passport_number": {"type": FIELD.PK_WITH_PREFIX, "zerofill": 6},
        },
        "public.misc_phone_borrowers": TABLE.SKIP,
    }
}

console.log(JSON.stringify(RULES, null, 4));

File: rules.py
import json
import sys
from utils import sqlgen, consts as c

RULES = {
    "tables": {
        "public.table1": {
            "first_name": {"type": c.FIELD_USING_PK, "prefix": "n"},
            "passport_seria": {"type": c.FIELD_USING_PK, "lpad": (4, '0')},
            "full_name": {"type": c.FIELD_USING_PK, "prefix": "n", "postfix": "d"},
        },
        "public.table2": {
            "__where": "idx != 5",
            "number": {"type": c.FIELD_USING_PK, "lpad": (6, ' '), "pk": "idx"},
            "flat": {"type": c.FIELD_REWRITE, "value": None},
        },
        "public.tmp1": c.TABLE_SKIP,
        "public.tmp2": c.TABLE_TRUNCATE,
    }
}

if '--json' in sys.argv:
    print json.dumps(RULES)
elif '--sql' in sys.argv:
    sqlgen.generate_sql(RULES['tables'])
